zerver/tests/test_upload.py

```diff
@@ -291,9 +338,8 @@ def test_file_download_unauthed(self) -> None:
 
         self.logout()
         response = self.client_get(uri)
-        self.assert_json_error(
-            response, "Not logged in: API authentication or user session required", status_code=401
-        )
+        self.assertEqual(response.status_code, 403)
+        self.assert_in_response("<p>You are not authorized to view this file.</p>", response)
```

Comment:
We're changing from the convention of returning json to return html here, is that desirable? (not going to confuse some clients etc.?)

Reply:
Hmm, not sure. Good point though. I want to render this when user tires to directly access a file which is not accessible. @timabbott thoughts?

Reply:
I think this is probably desirable for an endpoint that is expected to return a file, not a JSON response, but we should post in `#api design`.  Also let's do this change as its own independent prep commit, so we can give it the proper review attention it deserves.

Reply:
https://github.com/zulip/zulip/pull/20136#discussion_r795374873

Reply:
This change is due to new `{"override_api_url_scheme", "allow_anonymous_user_web"}),` in `/user_uploads` url. So, just that change as an independent commit doesn't make sense. The HTML response existed in `/user_uploads` before this PR, so that discussion is not part of this PR.